### ISSUE TITLE
Installation problem

### ISSUE THREAD

**Issue by cbas** [jviotti/idid#1]
Hi @jviotti -- great to see this utility exists. :smile:

I'm facing a problem installing idid.

There is a failure during `npm install` and `npm install -g idid` when there is no `~/_idid.json` present. Since this file is only created once idid runs the first time, it can not be installed without manual editing. This seems tricky for new users who just want to use the CLI app and get on with their work.

The problem seems to be caused by the tests which are in `package.json` under `scripts.prepublish`. For some weird reason npm is designed to run `prepublish` when you run `npm install` without arguments. See the discussion at https://github.com/npm/npm/issues/3059 where others face the same problem.

1. Can the `prepublish` hook please be removed/changed? There should be no reason to run lint/build/test during installation anyway since it consumes a lot of time. Not sure how this will impact your workflow so I didn't want to create a PR just yet.
1. Can the test be fixed to mock `~/_idid.json` correctly? I'm not familiar with `mock-fs` but it's not clearly mocking the file. (Could be a Windows problem?)

**Comment by cbas** [comment]
FYI as a workaround you can run:

    npm install -g --production idid

This will skip the `prepublish` step.

**Comment by jviotti** [comment]
Hi @cbas, thanks for reaching out. I didn't test this tool extensively on Windows, but let me play around maybe later today.

1. The purpose of the prepublish hook is to prevent publishing a version to NPM whose tests don't pass.
2. I'll definitely take a look at this.

What version of Windows and NodeJS are you running the tool on, so I have more chances of being able to reproduce it?

Also, if you have it at hand, can you paste the output of the download?

**Comment by cbas** [comment]
I was testing this on Windows 10 and the latest version of iojs. Will be
able to get exact information tomorrow.

**Comment by jviotti** [comment]
@cbas Great, looking forward to that information. In the mid time I've integrated Appveyor (http://www.appveyor.com) with NodeJS 0.10, 0.11 and 0.12, which succeeded without any error, which enforces our theories of either Windows 10 or iojs breaking mock-fs.

**Comment by jviotti** [comment]
Not sure if it's what you're getting, but was able to reproduce an issue in Appveyor with iojs 2.3: https://ci.appveyor.com/project/jviotti/idid/build/1.0.8/job/57vb85aydq8brvjw.

```sh
[16:09:58] Error in plugin 'gulp-mocha'
Message:
    ENOENT, no such file or directory '\\?\C:\projects\idid\lib\config.coffee'
Details:
    code: ENOENT
    errno: 34
    domain: [object Object]
    domainThrown: true
Stack:
Error: ENOENT, no such file or directory '\\?\C:\projects\idid\lib\config.coffee'
  at Binding.<anonymous> (C:\projects\idid\node_modules\mock-fs\lib\binding.js:302:15)
  at maybeCallback (C:\projects\idid\node_modules\mock-fs\lib\binding.js:41:17)
  at Binding.open (C:\projects\idid\node_modules\mock-fs\lib\binding.js:276:10)
  at Object.fs.openSync (C:\projects\idid\node_modules\mock-fs\node\fs-2.0.0.js:478:18)
  at Object.fs.readFileSync (C:\projects\idid\node_modules\mock-fs\node\fs-2.0.0.js:330:15)
  at Object.exports._compileFile (C:\projects\idid\node_modules\gulp-coffeelint\node_modules\coffeelint\node_modules\coffee-script\lib\coffee-script\coffee-script.js:224:14)
  at getSourceMap (C:\projects\idid\node_modules\gulp-coffeelint\node_modules\coffeelint\node_modules\coffee-script\lib\coffee-script\coffee-script.js:343:22)
  at getSourceMapping (C:\projects\idid\node_modules\gulp-coffeelint\node_modules\coffeelint\node_modules\coffee-script\lib\coffee-script\coffee-script.js:351:19)
  at formatSourcePosition (C:\projects\idid\node_modules\gulp-coffeelint\node_modules\coffeelint\node_modules\coffee-script\lib\coffee-script\coffee-script.js:303:16)
  at C:\projects\idid\node_modules\gulp-coffeelint\node_modules\coffeelint\node_modules\coffee-script\lib\coffee-script\coffee-script.js:369:33
  at Function.Error.prepareStackTrace (C:\projects\idid\node_modules\gulp-coffeelint\node_modules\coffeelint\node_modules\coffee-script\lib\coffee-script\coffee-script.js:372:7)
  at C:\projects\idid\node_modules\mocha\lib\reporters\base.js:167:20
  at Array.forEach (native)
  at Function.exports.list (C:\projects\idid\node_modules\mocha\lib\reporters\base.js:158:12)
  at Landing.Base.epilogue (C:\projects\idid\node_modules\mocha\lib\reporters\base.js:311:10)
  at Runner.<anonymous> (C:\projects\idid\node_modules\mocha\lib\reporters\landing.js:88:10)
  at emitNone (events.js:72:20)
  at Runner.emit (events.js:166:7)
  at C:\projects\idid\node_modules\mocha\lib\runner.js:596:10
  at C:\projects\idid\node_modules\mocha\lib\runner.js:517:7
  at next (C:\projects\idid\node_modules\mocha\lib\runner.js:246:23)
  at Immediate._onImmediate (C:\projects\idid\node_modules\mocha\lib\runner.js:275:5)
  at processImmediate [as _immediateCallback] (timers.js:371:17)
```

**Comment by jviotti** [comment]
@cbas I have a fix that works under Windows 8.1 and iojs v2.3.4. Can you confirm that the branch `fix/iojs-windows` also works for you?

**Comment by cbas** [comment]
Great! That fixed it on my Windows machine too. :+1:

**Comment by jviotti** [comment]
@cbas That's great to hear. I've just released v1.0.1 to NPM. Let me know if you run into any other issue, and thanks for using the tool!
